feat: Introduce Azure Event Grid as a new CloudEvent destination (#1336)
kedacore/keda-docs: content/docs/2.14/operate/events.md

  name: {cloud-event-name}
spec:
  clusterName: {cluster-name} #Optional. Will be used in the source/subject to specify where the event comes from. The default value is 'kubernetes-default' and it can also be set during the installation of KEDA with --k8sClusterName. This one will overwrite others if set.
  authenticationRef: 

>     name: {trigger-authentication-name} #Optional. Used to reference a `TriggerAuthentication` for authentication. 
>     kind: ClusterTriggerAuthentication #Optional. Used to choose the authenticatio scopes. https://keda.sh/docs/2.13/concepts/authentication/#authentication-scopes-namespace-vs-cluster

Reviewer: Let's use namespace-scoped TA by default
```suggestion
    name: {trigger-authentication-name} #Optional. Used to reference a `TriggerAuthentication` for authentication. 
    kind: TriggerAuthentication # Optional. Used to choose the authentication scopes. https://keda.sh/docs/latest/concepts/authentication/#authentication-scopes-namespace-vs-cluster
```

Is `name` really optional? I presume it is only when `azureEventGridTopic` is not used?

Author: It's optional. I reused the ta logic of ScaledObejct. So all properties requirement are the same as ScaledObejct.

  destination:
    http:
      uri: http://foo.bar

>     azureEventGrid:

Author: namespace isn't supported rightnow. To support namespace should use another lib. Topic and namespace are using two different libs.

>     azureEventGrid:

Reviewer: Exactly my point :) So let's clarify that in the name and add topic in there

      endpoint: https://my-topic.eastus-1.eventgrid.azure.net/api/events

>       key: ***

Reviewer: I believe this needs to be removed

Here is an overview of the supported destinations:

- [HTTP endpoint](#http-endpoint).
- [Azure Event Grid endpoint](#azure-event-grid).

#### HTTP endpoint
```yaml
  destination:
    http:
      uri: http://foo.bar  #An http endpoint that can receive cloudevent
```
#### Azure Event Grid
```yaml
  destination:
    azureEventGrid:

>       endPoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic

Reviewer: Let's fix the casing, can we also use a proper event grid topic example?
```suggestion
      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
```

>       key: *** #accesskey to  AzureEventGrid Topic

Reviewer: Let's update to use trigger authentication

Here is an overview of the supported destinations:

- [HTTP endpoint](#http-endpoint).
- [Azure Event Grid endpoint](#azure-event-grid).

#### HTTP endpoint
```yaml
  destination:
    http:
      uri: http://foo.bar  #An http endpoint that can receive cloudevent
```
#### Azure Event Grid
```yaml
  destination:
    azureEventGrid:
      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
```

> ```TriggerAuthentication``` is needed to provide accesskey or identity for Azure Evnet Grid authentication.

Reviewer: How do I link these two?

Author: KEDA will try to find if TA has a `key` value.

Reviewer: That is not how it should work though, the end-user should specify the trigger authentication resource to use or there is a security risk

**Connection String Authentication:**

> - `key` - accesskey string for the Azure Event Grid auth.

Reviewer: ```suggestion
- `key` - Access key string for the Azure Event Grid auth.
```

Is this conenctio nstring or?

Author: It's an access key that requested with endpoint. It's not a connection string.

      uri: http://foo.bar  #An http endpoint that can receive cloudevent
```

#### Azure Event Grid

> ```yaml

Reviewer: Nit
```suggestion

```yaml
```

  destination:
    azureEventGrid:
      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
```

> ```TriggerAuthentication``` is needed to provide accesskey or identity for Azure Event Grid authentication. ```authenticationRef``` should be configured when sinking to Azure Event Grid

Reviewer: ```suggestion

Authentication information must be provided by using `authenticationRef` which allows you to provide the access key or managed identity for Azure Event Grid authentication by providing a `TriggerAuthentication`.

Here is an overview of the supported authentication types:
```

> **Connection String Authentication:**

Reviewer: ```suggestion
##### Connection String Authentication
```

> - `accessKey` - Access key string for the Azure Event Grid connection auth.

Reviewer: To verify - `accessKey` is the expected key from TA, right?

Author: Yes. TA needs the `accessKey`. And KEDA will use it to auth with connection string.

> **Pod identity based authentication:**
> [Azure AD Workload Identity](https://azure.github.io/azure-workload-identity/docs/) providers can be used.
> 
> ```yaml

Reviewer: ```suggestion
##### Pod identity based authentication
[Azure AD Workload Identity](https://azure.github.io/azure-workload-identity/docs/) providers can be used.

```yaml
```

apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: nameOfTriggerAuth
  namespace: default
spec:
  podIdentity:

>     provider: azure-workload

Reviewer: Let's close the code sample
